contracts/libs/Types.sol

```diff
@@ -29,39 +29,102 @@ library Types {
         bytes32 depositRoot;
         bool withdrawn;
     }
+    struct Commitment {
+        bytes32 stateRoot;
+        bytes32 bodyRoot;
+    }
```

Comment:
To dispute a batch, we need the previous correct state root. With this Commitment struct, we can just get the state root without worrying if the commitment is a transfer commitment or Mass migration commitment.